meshroom/ui/components/thumbnail.py

```diff
@@ -0,0 +1,210 @@
+        """Load an image, resize it to thumbnail dimensions and save the result in the cache directory.
+
+        Args:
+            imgSource (QUrl): location of the input image
+        """
+        imgPath = imgSource.toLocalFile()
+        path = ThumbnailCache.thumbnailPath(imgPath)
+        logging.debug(f'[ThumbnailCache] Creating thumbnail {path} for image {imgPath}')
+
+        # Initialize image reader object
+        reader = QImageReader()
+        reader.setFileName(imgPath)
+        reader.setAutoTransform(True)
+
+        # Read image and check for potential errors
+        img = reader.read()
+        if img.isNull():
+            logging.error(f'[ThumbnailCache] Error when reading image: {reader.errorString()}')
+            return
+
+        # Scale image while preserving aspect ratio
+        thumbnail = img.scaled(ThumbnailCache.thumbnailSize, aspectMode=Qt.KeepAspectRatio)
+
+        # Write thumbnail to disk and check for potential errors
+        writer = QImageWriter(path)
+        success = writer.write(thumbnail)
+        if not success:
+            logging.error(f'[ThumbnailCache] Error when writing thumbnail: {writer.errorString()}')
+
+        # Notify listeners
+        self.thumbnailCreated.emit(imgSource)
+
+    @staticmethod
+    def clean():
+        """Scan the thumbnail directory and:
+        1. remove all thumbnails that have not been used for more than storageTimeLimit days
+        2. ensure that the number of thumbnails on disk does not exceed maxThumbnailsOnDisk.
+        """
+        # Check if thumbnail directory exists
+        if not os.path.exists(ThumbnailCache.thumbnailDir):
+            logging.debug('[ThumbnailCache] Thumbnail directory does not exist yet.')
+            return
+
+        # Get current time
+        now = time.time()
+
+        # Scan thumbnail directory and gather all thumbnails to remove
+        toRemove = []
+        remaining = []
+        for entry in os.scandir(ThumbnailCache.thumbnailDir):
+            if not entry.is_file():
+                continue
+
+            # Compute storage duration since last usage of thumbnail
+            lastUsage = os.path.getmtime(entry.path)
+            storageTime = now - lastUsage
+            logging.debug(f'[ThumbnailCache] Thumbnail {entry.name} has been stored for {storageTime}s')
+
+            if storageTime > ThumbnailCache.storageTimeLimit * 3600 * 24:
+                # Mark as removable if storage time exceeds limit
+                logging.debug(f'[ThumbnailCache] {entry.name} exceeded storage time limit')
+                toRemove.append(entry.path)
+            else:
+                # Store path and last usage time for potentially sorting and removing later
+                remaining.append((entry.path, lastUsage))
+
+        # Remove all thumbnails marked as removable
+        for path in toRemove:
+            logging.debug(f'[ThumbnailCache] Remove {path}')
+            os.remove(path)
```

Comment:
The various call to `os.remove` could raise if the file does not exist (for instance another instance of Meshroom).
The whole function should be in a try/except FileNotFoundError.